tests/lib/devicetree/emul/src/emulator.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+
+static int emul_init(const struct emul *emul, const struct device *parent)
+{
+	return 0;
+}
+
+/*
+ * This emulator definition is deliberately in a different compilation unit than
+ * test_emuL_dt_get to verify that emulators are externally accessible.
+ */
+EMUL_DEFINE(emul_init, DT_NODELABEL(test_gpio_0), NULL, NULL);
```

Comment:
Can we include an `EMUL_DT_DEFINE` to mirror `DEVICE_DT_DEFINE`. The contract of `EMUL_DT_GET` should be that it will map 1:1 with `EMUL_DT_DEFINE`, if the developer used anything else it's on them to make sure that the IDs are aligned. The appropriate test should include:

1. A new binding with a test `compatible` string
2. The test `emulator.c` defining `DT_DRV_COMPAT` to match the compatible string
3. The test `emulator.c` calling `DT_INST_FOREACH_STATUS_OKAY` to initiate one emulator per instance of `DT_DRV_COMPAT`
4. The test `main.c` should find the instance using any of the node identifier get macros (`DT_NODELABEL` is one of them so that works like you have it).